src/rgw/rgw_sal.h

```diff
@@ -1498,20 +1499,29 @@ class Zone {
 };
 
 /**
- * @brief Abstraction of a manager for Lua scripts
+ * @brief Abstraction of a manager for Lua scripts and packages
  *
- * RGW can load and process Lua scripts.  This will handle loading/storing scripts.
+ * RGW can load and process Lua scripts.  This will handle loading/storing
+ * scripts; adding, deleting, listing, and installing packages.
  */
-class LuaScriptManager {
+class LuaManager {
 public:
-  virtual ~LuaScriptManager() = default;
+  virtual ~LuaManager() = default;
 
   /** Get a script named with the given key from the backing store */
-  virtual int get(const DoutPrefixProvider* dpp, optional_yield y, const std::string& key, std::string& script) = 0;
+  virtual int get_script(const DoutPrefixProvider* dpp, optional_yield y, const std::string& key, std::string& script) = 0;
   /** Put a script named with the given key to the backing store */
-  virtual int put(const DoutPrefixProvider* dpp, optional_yield y, const std::string& key, const std::string& script) = 0;
+  virtual int put_script(const DoutPrefixProvider* dpp, optional_yield y, const std::string& key, const std::string& script) = 0;
   /** Delete a script named with the given key from the backing store */
-  virtual int del(const DoutPrefixProvider* dpp, optional_yield y, const std::string& key) = 0;
+  virtual int del_script(const DoutPrefixProvider* dpp, optional_yield y, const std::string& key) = 0;
+  /** Add a lua package */
```

Comment:
why not protect these under the `ifdef`?

Reply:
it keeps the interface the same regardless of how it's built.

I thought that was conceptually cleaner. Theoretically less likely to cause strange crashes if somehow you get a mismatched rgw and plugin compiled different ways, i.e.  radosgw compiled *with* the ifdef, but plugin compiled *without* the ifdef

Reply:
It should be okay.  Anything overridden will call the overriding functions, anything not overridden will use the base implementation.

Reply:
but this will work only if there is a base implementation.
if we keep these functions as pure virtuals that won't compile